### PR TITLE
fix: upgrade knex from 2.4.0 to 2.4.2

### PR DESCRIPTION
**Pull request by owlas**
Also fixes new warnings with double rollbacks in transaction blocks

2.4.0 contains a bug when inserting arrays: https://github.com/knex/knex/issues/5365

### Reviewing

The diff looks big but I've only made two changes:

 - upgrading to 2.4.2
 - Removing the `try {} except {trx.rollback(e); throw e;}` from a bunch of transacting blocks

### CODE REVIEW

File: packages/backend/src/models/DashboardModel/DashboardModel.ts [hunk 1]
```diff
@@ -635,30 +635,25 @@ export class DashboardModel {
...
-                            .update({
-                                name: dashboard.name,
-                                description: dashboard.description,
-                                ...withSpaceId,
-                            })
-                            .where('dashboard_uuid', dashboard.uuid);
-                    }),
-                );
-            } catch (e) {
-                trx.rollback(e);
```
Comment (rephus) [inline packages/backend/src/models/DashboardModel/DashboardModel.ts:659]: why don't we want those rollbacks ? are they applied automatically ?

**Review by ZeRego — APPROVED** [review]

**Review by rephus — APPROVED** [review]
IF e2e pass, it should be safe

File: packages/backend/src/models/DashboardModel/DashboardModel.ts (hunk 1 above)
Comment (ZeRego) [reply packages/backend/src/models/DashboardModel/DashboardModel.ts:659]: yes, knex handles that for us. This was a bad practice when we started using knex.

Comment (owlas) [reply packages/backend/src/models/DashboardModel/DashboardModel.ts:659]: That's right! About half way down this page: https://knexjs.org/guide/transactions.html

And in dev when the error happened it was inside this transaction block. And the knex threw another error (hiding the real one) saying "you're trying to exit a transaction that has already been closed" (or something like that).

This fixed the dev experience. It doesn't seem to happen in production though, which is curious

**Pull request merged**